### PR TITLE
[4.0] Switch usage of inputField.onchange in calendar fields to use dispatc…

### PR DESCRIPTION
**Pull request by GeraintEdwards**
…hEvent(new Event('change' - this ensures scripts that watch the input field using eventListeners and onchange attributes will respond to the event.

Pull Request for Issue # .

### Summary of Changes

changes in calendar field javascript to mirror  https://github.com/joomla/joomla-cms/pull/33540  and https://github.com/joomla/joomla-cms/pull/32756

### Testing Instructions

1. Testing onchange handler

edit the file administrator/components/com_content/forms/article.xml and find the publish_up field at line 129 and add the attrribute

onchange="alert(1);"

to this field and then edit an article and change the publish_up field (using the date picker)

2. Testing change event listener

Add to Atum index.php before closing </body> tag :
```

<script type="module">
if (window.jform_publish_up){
  window.jform_publish_up.addEventListener('change', () => {
    alert(2);
  })
}
</script>

```

Go to Article editing, and try to change the publish_up date.

### Actual result BEFORE applying this Pull Request

You get a two alert(1) messages and one alert(2)

### Expected result AFTER applying this Pull Request

You get one alert(1) message and alert(2)

### CODE REVIEW

**Comment by richard67** [comment]
@GeraintEdwards Your PR starts with a conflict in file `build/media_source/system/js/fields/calendar.es5.js`. Can it be that your 4.0-dev branch was outdated when you created that PR, or is that a while ago?

**Comment by GeraintEdwards** [comment]
Thanks @richard67 - conflict resolved now.  Don't know why by fetch upstream and merge didn't work first time!

File: build/media_source/system/js/fields/calendar.es5.js [hunk 1]
```diff
@@ -234,9 +234,7 @@
 		}
 		this.inputField.value = this.date.print(this.params.dateFormat, this.params.dateType, true);
 
-		if (typeof this.inputField.onchange == "function") {
-			this.inputField.onchange();
-		}
+		this.inputField.dispatchEvent(new Event('change', {bubbles: true, cancelable: true}));
```
Comment (Fedik) [inline build/media_source/system/js/fields/calendar.es5.js]: please remove it,
this already exists a few lines down

Comment (GeraintEdwards) [reply build/media_source/system/js/fields/calendar.es5.js]: done - more haste less speed!

**Comment by ceford** [comment]
Not sure I understand this one. I get two (1) alerts and on (2) alert with or without the patch. I have checked that npm run build:js gives the correct js output. I must have done something wrong in the test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33731">issues.joomla.org/tracker/joomla-cms/33731</a>.</sub>

**Comment by GeraintEdwards** [comment]
@ceford 
> Not sure I understand this one. I get two (1) alerts and on (2) alert with or without the patch. I have checked that npm run build:js gives the correct js output. I must have done something wrong in the test.

My mistake this PR mirrored one from Joomla 3.9.x and I had initially miss-read the code and added a duplicated dispatchEvent call by mistake.  I did correct the file about 5 hours ago - I don't know if you had tested the patch before then or if the patch tester has a delay on it??  

I have amended the testing instructions too now.

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on d97b8530a16eff012d0752a98fa7069ba7655047<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33731">issues.joomla.org/tracker/joomla-cms/33731</a>.</sub>

**Comment by GeraintEdwards** [comment]
This PR can be closed since #5b54c98b762df1f342eb5c374cc7dff15d86bd3b on 22/05 merging in staging applied the same changes

**Pull request closed without merging**